src/librustdoc/html/render/context.rs

```diff
@@ -84,10 +91,6 @@ crate struct SharedContext<'tcx> {
     /// This describes the layout of each page, and is not modified after
     /// creation of the context (contains info like the favicon and added html).
     crate layout: layout::Layout,
-    /// This flag indicates whether `[src]` links should be generated or not. If
-    /// the source files are present in the html rendering, then this will be
-    /// `true`.
-    crate include_sources: bool,
```

Comment:
Why did this move?

Reply:
The biggest reason is this one:

```
error[E0594]: cannot assign to data in an `Rc`
   --> src/librustdoc/html/sources.rs:117:13
    |
117 | /             self.cx.shared.include_sources = match self.emit_source(&filename, start_pos.0) {
118 | |                 Ok(()) => true,
119 | |                 Err(e) => {
120 | |                     self.cx.shared.tcx.sess.span_err(
...   |
125 | |                 }
126 | |             };
    | |_____________^ cannot assign
    |
    = help: trait `DerefMut` is required to modify through a dereference, but it is not implemented for `Lrc<SharedContext<'_>>`
```

The second advantage is that it reduces the size of `SharedContext` too.